test/integration/targets/user/tasks/main.yml

```diff
@@ -429,6 +429,52 @@
           - bsd_account_expiration.stdout == '0'
   when: ansible_os_family == 'FreeBSD'
 
+# Test setting epoch 0 expiration when creating a new account, then removing the expiry
+# https://github.com/ansible/ansible/issues/47114
+- name: Remove ansibulluser
+  user:
+    name: ansibulluser
+    state: absent
+
+- name: Create user account with epoch 0 expiration
+  user:
+    name: ansibulluser
+    state: present
+    expires: 0
+
+- name: Change the user account to remove the expiry time
+  user:
+    name: ansibulluser
+    expires: -1
```

Comment:
Run this again to test no changes are report. Currently, it always reports a change.